Gcpe.Hub.API.IntegrationTests/Views/TestMessagesViews.cs

```diff
@@ -1,116 +1,108 @@
 
-            Assert.NotEmpty(deserializedBody);
-            deserializedBody.Should().HaveCountGreaterOrEqualTo(5);
+            for (int i = 0; i < models.Count() - 1; i++)
+            {
+                models.ElementAt(i).SortOrder.Should().BeLessThan(models.ElementAt(i + 1).SortOrder);
+            }
         }
 
         [Fact]
         public async Task Create_EndpointReturnSuccessAndCorrectMessage()
         {
-            testMessage.Id = Guid.Empty;
-            var stringContent = new StringContent(JsonConvert.SerializeObject(testMessage), Encoding.UTF8, "application/json");
-
-            var response = await _client.PostAsync("/api/messages",  stringContent);
-            response.EnsureSuccessStatusCode();
-            var body = await response.Content.ReadAsStringAsync();
-            var messageResult = JsonConvert.DeserializeObject<Hub.API.ViewModels.MessageViewModel>(body);
+            var newMessage = TestData.CreateSerializedMessage("Test title!", "test description!", 0, true, false);
+            var createResponse = await _client.PostAsync("/api/messages", newMessage);
+            createResponse.EnsureSuccessStatusCode();
+            var createBody = await createResponse.Content.ReadAsStringAsync();
+            var messageResult = JsonConvert.DeserializeObject<MessageViewModel>(createBody);
 
-            messageResult.Title.Should().Be(testMessage.Title);
-            messageResult.Description.Should().Be(testMessage.Description);
+            messageResult.Title.Should().Be("Test title!");
+            messageResult.Description.Should().Be("test description!");
         }
 
         [Fact]
         public async Task Create_EndpointRequiresTitle()
         {
-            testMessage.Id = Guid.Empty;
-            var brokenTestMessage = testMessage;
-            brokenTestMessage.Title = null;
-            var stringContent = new StringContent(JsonConvert.SerializeObject(brokenTestMessage), Encoding.UTF8, "application/json");
-
-            var response = await _client.PostAsync("/api/messages", stringContent);
+            var brokenTestMessage = TestData.CreateSerializedMessage(null, "description", 0, true, false);
+            var response = await _client.PostAsync("/api/messages", brokenTestMessage);
             response.IsSuccessStatusCode.Should().BeFalse();
         }
 
         [Fact]
         public async Task Create_EndpointDoesntRequireDescription()
         {
-            testMessage.Id = Guid.Empty;
-            var noDescMessage = testMessage;
-            noDescMessage.Description = null;
-            var stringContent = new StringContent(JsonConvert.SerializeObject(noDescMessage), Encoding.UTF8, "application/json");
+            var noDescMessage = TestData.CreateSerializedMessage("Title", null, 0, true, false);
 
-            var response = await _client.PostAsync("/api/messages", stringContent);
+            var response = await _client.PostAsync("/api/messages", noDescMessage);

```

Comment:
Reuse function above

Reply:
For tests that are testing the Post views directly I opted to not reuse the helper function so that it is more clear what is being tested.